### PR TITLE
Add IP Video support

### PR DESCRIPTION
**Pull request by JoshuaWatt**
**Description**

Adds support for streaming video from IP video sources such as an HDMI
to IP capture device or an IP camera.

**Checklist**
- [x] Documentation for the feature
- [x] Tests for the feature 

- [x] The arguments and description in doc/configuration.rst have been updated

- [ ] Add a section on how to use the feature to doc/usage.rst

- [ ] Add a section on how to use the feature to doc/development.rst

- [ ] CHANGES.rst has been updated

- [x] PR has been tested

- [ ] Man pages have been regenerated

### CODE REVIEW

**Review by jluebbe — CHANGES_REQUESTED** [review]
Looks good otherwise. Did you see #759? I add V4L controls there, which would be incompatible with this. I could drop https://github.com/labgrid-project/labgrid/pull/759/commits/0d06401cfe6ec57dface83340561f6b3d0030f87 from that PR and discuss that separately.

File: doc/configuration.rst [hunk 1]
```diff
@@ -797,6 +797,26 @@ Such device could be a signal generator.
...
+
+Used by:
+  - `IPVideoDriver`_
+
+NetworkIPVideo
+~~~~~~~~~~~~~~~
+
+A :any:`NetworkIPVideo` resource describes a :any:`IPVideo` resource available
+on a remote computer.
+
```
Comment (jluebbe) [inline doc/configuration.rst:819]: It seem GitHub lost my comment...
Only IPVideo (maybe named HTTPVideoStream?) would be needed, similar to ModbusTCPCoil or OneWirePIO, where the exporter is not involved and just passes through the information.

Comment (JoshuaWatt) [reply doc/configuration.rst:819]: Ah, I though the Network version would be needed for SSH proxying in isolated mode so I included it, but maybe it's not involved in that code path?

Comment (jluebbe) [reply doc/configuration.rst:819]: It shouldn't be, the proxy information (host and required) is added in the 'extra' subkey and not part of the resource attributes.

**Comment by JoshuaWatt** [comment]
> Looks good otherwise. Did you see #759? I add V4L controls there, which would be incompatible with this. I could drop [0d06401](https://github.com/labgrid-project/labgrid/commit/0d06401cfe6ec57dface83340561f6b3d0030f87) from that PR and discuss that separately.

I didn't see that. It might be fine to add that to the `VideoProcotol::stream()` function anyway

File: labgrid/driver/usbvideodriver.py [hunk 2]
```diff
@@ -5,15 +5,16 @@
...
 @target_factory.reg_driver
 @attr.s(eq=False)
-class USBVideoDriver(Driver):
+class USBVideoDriver(Driver, VideoProtocol):
     bindings = {
         "video": {"USBVideo", "NetworkUSBVideo"},
     }
 
-    def get_caps(self):
+    def get_qualities(self):
```
Comment (Emantor) [inline labgrid/driver/usbvideodriver.py:17]:
```
Traceback (most recent call last):
  File "/home/phoenix/work/ptx/labgrid/labgrid/remote/client.py", line 1798, in main
    args.func(session)
  File "/home/phoenix/work/ptx/labgrid/labgrid/remote/client.py", line 1161, in video
    drv.stream(quality)
  File "/home/phoenix/work/ptx/labgrid/labgrid/binding.py", line 100, in wrapper
    return func(self, *_args, **_kwargs)
  File "/home/phoenix/work/ptx/labgrid/labgrid/driver/usbvideodriver.py", line 52, in stream
    caps = self.select_caps(caps_hint)
  File "/home/phoenix/work/ptx/labgrid/labgrid/driver/usbvideodriver.py", line 34, in select_caps
    default, variants = self.get_caps()
AttributeError: 'USBVideoDriver' object has no attribute 'get_caps'
```

**Review by Emantor — CHANGES_REQUESTED** [review]

**Comment by JoshuaWatt** [comment]
Ping?

**Review by jluebbe — APPROVED** [review]

File: labgrid/driver/usbvideodriver.py (hunk 2 above)
Comment (jluebbe) [reply labgrid/driver/usbvideodriver.py:17]: This was resolved by renaming all `get_caps` to `get_qualities`.

**Pull request merged**